src/screens/LunchScreen.js

```diff
@@ -0,0 +1,155 @@
+import {
+  BoldText,
+  Button,
+  ConnectedImagesCarousel,
+  HeaderLeft,
+  Icon,
+  LunchSection,
+  RegularText,
+  SafeAreaViewFlex,
+  Wrapper,
+  WrapperRow,
+  WrapperWithOrientation
+} from '../components';
+import { arrowLeft, arrowRight } from '../icons';
+import { useMatomoTrackScreenView } from '../hooks';
+import { graphqlFetchPolicy } from '../helpers';
+import moment from 'moment';
+import { NetworkContext } from '../NetworkProvider';
+import { useQuery } from 'react-apollo';
+import { getQuery, QUERY_TYPES } from '../queries';
+
+const { MATOMO_TRACKING } = consts;
+
+export const LunchScreen = ({ navigation }) => {
+  const [poiId, setPoiId] = useState(navigation.getParam('poiId'));
+
+  const [date, setDate] = useState(moment());
+
+  const { isConnected, isMainserverUp } = useContext(NetworkContext);
+  const fetchPolicy = graphqlFetchPolicy({ isConnected, isMainserverUp });
+
+  const currentDate = moment(date).format('YYYY-MM-DD');
+
+  const variables = {
+    dateRange: [currentDate, currentDate]
+  };
+
+  const { data, loading, refetch } = useQuery(getQuery(QUERY_TYPES.LUNCHES), {
+    fetchPolicy,
+    variables
+  });
+
+  useMatomoTrackScreenView(MATOMO_TRACKING.SCREEN_VIEW.LUNCH);
+
+  const onPressNext = useCallback(() => {
+    setDate((oldDate) => moment(oldDate).add(1, 'day'));
+  }, [setDate]);
+
+  const onPressPrevious = useCallback(() => {
+    setDate((oldDate) => moment(oldDate).subtract(1, 'day'));
+  }, [setDate]);
+
+  const renderItem = useCallback(
+    ({ item }) => <LunchSection lunchOfferData={item} navigation={navigation} />,
+    [navigation]
+  );
+
+  let lunchData = data?.[QUERY_TYPES.LUNCHES];
+
+  if (poiId) {
+    lunchData?.filter((item) => item?.pointOfInterest?.id === poiId);
+  }
+
+  const ListHeaderComponent = (
+    <>
+      <ConnectedImagesCarousel
+        navigation={navigation}
+        publicJsonFile="lunchCarousel"
+        refreshTimeKey="publicJsonFile-lunchCarousel"
+      />
+      <WrapperWithOrientation>
+        <Wrapper>
+          <WrapperRow>
+            <TouchableOpacity onPress={onPressPrevious} style={styles.left}>
+              <Icon xml={arrowLeft(colors.primary)} />
+            </TouchableOpacity>
+            <BoldText big>{date.format('DD.MM.YYYY')}</BoldText>
+            <TouchableOpacity onPress={onPressNext} style={styles.right}>
+              <Icon xml={arrowRight(colors.primary)} />
+            </TouchableOpacity>
+          </WrapperRow>
+        </Wrapper>
+      </WrapperWithOrientation>
+    </>
+  );
+
+  const ListEmptyComponent = (
+    <Wrapper>
+      <RegularText>{texts.lunch.noOffers}</RegularText>
+    </Wrapper>
+  );
+
+  const ListFooterComponent = (
+    <Wrapper>
+      <Button title={texts.lunch.showAll} onPress={() => setPoiId(undefined)} />
+    </Wrapper>
+  );
+
+  return (
+    <SafeAreaViewFlex>
+      <FlatList
+        refreshControl={
+          <RefreshControl
```

Comment:
is the `refreshing` prop missing, which we have on `HomeScreen` for example and seems to be required by `RefreshControl`?

![Bildschirmfoto 2021-02-09 um 19 05 18](https://user-images.githubusercontent.com/1942953/107407350-d9ed0900-6b09-11eb-8493-b123e8f32b75.png)
https://reactnative.dev/docs/refreshcontrol#refreshing

Reply:
you are right, it is missing there. i did leave it out on purpose, not realizing it was flagged as required.

i added a `refreshing={false}` now.

using `loading` there causes the ActivityIndicator to always animate in from the top and pushing everything down under ios, whenever you swap the date, so i left it out.